### ISSUE TITLE
Additional exception(s) for improper use of link

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1432]
[This website](https://iowalakes.edu/campus-life/music/) has an audio player on it that uses a link `<a>` with `role="slider"` to control the volume.

<img width="841" height="419" alt="screenshot of front end and dev tools code view on the media player" src="https://github.com/user-attachments/assets/3cb12c18-fc9e-40f0-b675-e0f5f9636049" />

This is the code:
```
<a class="mejs-horizontal-volume-slider" href="javascript:void(0);" aria-label="Volume Slider" aria-valuemin="0" aria-valuemax="100" aria-valuenow="80" role="slider" style="display: block;" aria-valuetext="80%"><span class="mejs-offscreen">Use Up/Down Arrow keys to increase or decrease volume.</span><div class="mejs-horizontal-volume-total"><div class="mejs-horizontal-volume-current" style="left: 0px; width: 80%;"></div><div class="mejs-horizontal-volume-handle" style="left: 80%;"></div></div></a>
```

This flags an "improper use of link" error. Normally, a [slider role would be applied to a div](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/slider_role), but in this example, on an `<a>`, it sounds OK on a screen reader. It announces correctly as a slider, not a link, and it can be interacted with using the arrow keys. (There's a different issue here RE the instructions not being announced, but that's not relevant at the moment.)

It seems like `role="slider"` should be an exception to the Improper use of link rule, but I am also wondering if the presence of any role (aside from link), should create a pass. Or, potentially, instead of returning Improper use of link as an error, should there be a "suspicious link" warning/review item?

For example, this currently flags an error:
`<a role="presentation">link with presentation role</a>`

Should it flag a warning to check it instead? It wouldn't be keyboard focusable and wouldn't read as a link to a screen reader, but it might still confuse sighted people because it would be styled like a link, so it would appear broken.

The customer-reported slider issue is easy to address, but @SteveJonesDev @pattonwebz, we might want to discuss this rule more broadly.

**Comment by amberhinds** [comment]
[These are all the possible role attributes.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles) I do want to say something to people if they're using random roles on links, because why are they using an anchor tag?? But I'm not sure where the line is.

**Comment by SteveJonesDev** [comment]
One option worth considering: rather than adding one-off exceptions as these cases surface, we could modify the rule to **pass when the `<a>` has a recognized interactive widget role** — `button`, `slider`, `checkbox`, `radio`, `menuitem`, `tab`, `switch`, `spinbutton`, `option`, `treeitem`, etc.

The principle: when a developer has explicitly overridden an `<a>`'s semantics with an interactive widget role (and included appropriate supporting ARIA attributes), they've addressed the concern the rule exists to catch. Using `<a>` as a natively focusable container and overriding the role is a known pattern — MediaElement.js isn't the only library doing it.

This keeps the scoping intentional. "Any non-link role = pass" would be too broad — `role="presentation"` on a link is still problematic and should still flag. But interactive widget roles are a coherent category with a clear rationale.

The alternative is treating each case as an edge case and adding exceptions one at a time, which will get fragile quickly.

**Comment by amberhinds** [comment]
> modify the rule to pass when the <a> has a recognized interactive widget role 

I think this is a good idea.
